packages/gatsby-plugin-sharp/src/index.js

```diff
@@ -395,9 +395,10 @@ async function traceSVG(args) {
 }
 
 async function stats({ file, reporter }) {
+  const pluginOptions = getPluginOptions()
   let imgStats
   try {
-    const pipeline = sharp()
+    const pipeline = sharp({ failOn: pluginOptions.failOn })
```

Comment:
Great, passing `failOn` down to `sharp` makes sense.

How about also passing the `pluginOptions` down to `reportErrors` so we can control the behavior of `reportError` (not to exit the whole process)?